hazelcast/src/main/java/com/hazelcast/sql/SqlRowMetadata.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql;
+
+import javax.annotation.Nonnull;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * SQL row metadata.
+ */
+public class SqlRowMetadata implements Serializable {
+    /** Constant indicating that the column is not found. */
+    public static final int COLUMN_NOT_FOUND = -1;
+
+    private static final long serialVersionUID = 1595632576318481583L;
+
+    private final List<SqlColumnMetadata> columns;
+    private final Map<String, Integer> nameToIndex;
+
+    public SqlRowMetadata(List<SqlColumnMetadata> columns) {
+        assert columns != null && !columns.isEmpty();
+
+        this.columns = new ArrayList<>(columns);
+
+        nameToIndex = new HashMap<>(columns.size());
+
+        for (int i = 0; i < columns.size(); i++) {
+            nameToIndex.put(columns.get(i).getName(), i);
+        }
+    }
+
+    /**
+     * Gets the number of columns in the row.
+     *
+     * @return The number of columns in the row.
+     */
+    public int getColumnCount() {
+        return columns.size();
+    }
+
+    /**
+     * Get column metadata.
+     *
+     * @param index Column index, 0-based.
+     * @return Column metadata.
+     */
+    @Nonnull
+    public SqlColumnMetadata getColumn(int index) {
+        if (index < 0) {
+            throw new IndexOutOfBoundsException("Column index cannot be negative: " + index);
+        }
+
+        if (index >= columns.size()) {
+            throw new IndexOutOfBoundsException("Column index is out of bounds: " + index);
+        }
+
+        return columns.get(index);
```

Comment:
These checks are perfectly handled by the list implementation, throw the same type of exception. 
```suggestion
        return columns.get(index);
```
We should also document that we throw in case of and out-of-bounds index, some other methods return null.

Reply:
As discussed, these checks are intentional to make the API behavior stable irrespective of the underlying implementation.